Carnap-Server/Foundation.hs

```diff
@@ -408,6 +411,26 @@ instance YesodAuth App where
                           String email <- HM.lookup "email" o
                           Just creds0 { credsIdent = email }
 
+            -- I want to make this multiline reasonably but ghci does not like it?!
+              noGoogleMessage = unwords ["Creating new Google accounts is disabled on "
+                                , "this Carnap instance. Log in via your "
+                                , "learning management system instead."]
+
+              tryCreateUser :: Text -> (YesodDB App (AuthenticationResult App))
```

Comment:
Possibly too slick - is there a way to write this more readably without MaybeT?

Reply:
You're definitely right, that control structure was unreasonably nonlinear. I've changed it to be much more explicit and clear using bind and a LambdaCase.